File: PULSE.Model/Requests/StaffInsertRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PULSE.Model.Requests
{
    public class StaffInsertRequest
    {
        [Required(AllowEmptyStrings = false)]
        public string FirstName { get; set; }
        [Required(AllowEmptyStrings = false)]
        public string LastName { get; set; }

        [Required(AllowEmptyStrings = false)]
        [EmailAddress()]
        public string Email { get; set; }
        [Required(AllowEmptyStrings = false)]
        public string PhoneNumber { get; set; }

        [MinLength(4)]
        [Required(AllowEmptyStrings = false)]
        public string Username { get; set; }
        [MinLength(4)]
        [Required(AllowEmptyStrings = false)]
        public string Password { get; set; }

        public string PasswordConfirm { get; set; }

        public DateTime DateOfBirth { get; set; }
        public DateTime DateOfEmployment { get; set; }

        public int RoleID { get; set; }
    }
}

File: pulse_admin/lib/api/types/user/requests/staff_insert_request.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pulse_admin/api/types/user/role.dart';
import 'package:pulse_admin/domain/types/forms/staff_form_field_names.dart';

part 'staff_insert_request.g.dart';

@JsonSerializable()
class StaffInsertRequest {
  String? firstName;
  String? lastName;
  String? username;
  String? email;
  String? password;
  String? passwordConfirm;
  int? roleID;
  String? phoneNumber;
  DateTime? dateOfBirth;
  DateTime? dateOfEmployment;

  StaffInsertRequest();

  factory StaffInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$StaffInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$StaffInsertRequestToJson(this);

  StaffInsertRequest.fromFormData(Map<String, dynamic> formData) {
    firstName = formData[StaffFormFieldNames.firstName];
    lastName = formData[StaffFormFieldNames.lastName];
    username = formData[StaffFormFieldNames.username];
    email = formData[StaffFormFieldNames.email];
    password = formData[StaffFormFieldNames.password];
    passwordConfirm = formData[StaffFormFieldNames.passwordConfirm];
    roleID = (formData[StaffFormFieldNames.role] as Role).id;
    phoneNumber = formData[StaffFormFieldNames.phoneNumber];
    dateOfBirth = formData[StaffFormFieldNames.dateOfBirth];
    dateOfEmployment = formData[StaffFormFieldNames.dateOfEmployment];
    print("RoleID: $roleID");
  }
}
